CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLHarvester.cc

```diff
@@ -615,7 +659,32 @@ SiPixelLAHarvest::fitResults SiPixelLorentzAnglePCLHarvester::fitAndStore(
   f1_->SetParError(5, 0);
   f1_->SetChisquare(0);
 
-  hists_.h_mean_[i_index]->getTH1()->Fit(f1_.get(), "ERQ");
+  TFitResultPtr result = hists_.h_mean_[i_index]->getTH1()->Fit(f1_.get(), "ERQS");
+  if (result.Get()) {
+    res.fitStatus = result->Status();
+    edm::LogInfo("SiPixelLorentzAnglePCLHarvester") << "Fit status: " << res.fitStatus;
+  } else {
+    edm::LogError("SiPixelLorentzAnglePCLHarvester") << "Could not retrieve the fit status! Setting it to 0 by default";
+    res.fitStatus = 0;
+  }
+
+  if (res.fitStatus != 0) {
+    res.covMatrixStatus = result->CovMatrixStatus();
+  } else {
+    res.covMatrixStatus = SiPixelLAHarvest::kNotCalculated;
+  }
+
+  // compute the error on the drift-at-half-width parameter (d^2)
+  float dSq{0.};
+  if (res.covMatrixStatus != SiPixelLAHarvest::kNotCalculated) {
+    for (int k = 0; k < order_; k++) {
+      for (int l = 0; l < order_; l++) {
+        dSq += (std::pow(half_width, k) * std::pow(half_width, l) * result->CovMatrix(k, l));
```

Comment:
From the formula which Morris shared, I understood this expression to be valid only for the standard polynomial of order `order_`. Is it also valid for Chebyshev polynomials?

Reply:
For safety in case of Chebyshev polynomial I've set it 0 in [490e6ae](https://github.com/cms-sw/cmssw/pull/38996/commits/490e6ae7e493e99ddb5a9259f92c65aa0d4120d4). In any case by default we won't run with that configuration.